### PR TITLE
Add amp-border-box class for use on html

### PR DESCRIPTION
**Pull request by cogell**
Quick pass at: https://github.com/ampproject/amphtml/issues/1497

### CODE REVIEW

**Comment by cogell** [comment]
I signed it!

**Comment by ampsauce** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 1a694315b5e237d9769cb5ceff2b3c277c048224
Build details: https://travis-ci.org/ampsauce/amphtml/builds/103951369

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: css/amp.css [hunk 1]
```diff
@@ -287,6 +287,17 @@ template {
...
+/**
+ * Provide common border-box sizing pattern to authors
+ */
+html.amp-border-box {
+  box-sizing: border-box;
+}
+html.amp-border-box *,
+html.amp-border-box *:before,
+html.amp-border-box *:after {
+  box-sizing: inherit;
```
Comment (dvoytenko) [inline css/amp.css]: Hmm. By default though `box-sizing` is not inherited. This could cause some unexpected behavior where someone would override box-sizing just for one element and it'd automatically apply to everything within.

What about if we just explicitly state `border-box` instead of `inherit`?

Comment (cogell) [inline css/amp.css]: Great point. My thinking was that html components that are built around a different `box-sizing` model would only need to set their `box-sizing` on the parent which would inherit down. Now I'm thinking that that case in an AMP page would be rare and in that edge case the author could assign `box-sizing` on each element on that component.

Will update.

**Comment by ampsauce** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 6ae0bfc3488d64002997620d300f7461e37bd10c
Build details: https://travis-ci.org/ampsauce/amphtml/builds/104164594

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by dvoytenko** [comment]
@cogell sorry, not sure, is this already updated and ready to be reviewed?

**Comment by cogell** [comment]
@dvoytenko sorry forget to ping. Yes it is updated. looks like the CI tests failed for a connection issue.

**Comment by cogell** [comment]
Should I add this new css class in the `docs/include_features.md` docs?

File: css/amp.css [hunk 2]
```diff
@@ -287,6 +287,17 @@ template {
   display: none !important;
 }
 
+/**
+ * Provide common border-box sizing pattern to authors
+ */
+html.amp-border-box {
+  box-sizing: border-box;
+}
+html.amp-border-box *,
```
Comment (dvoytenko) [inline css/amp.css]: nit: new line between declarations

File: css/amp.css [hunk 3]
```diff
@@ -287,6 +287,17 @@ template {
   display: none !important;
 }
 
+/**
+ * Provide common border-box sizing pattern to authors
```
Comment (dvoytenko) [inline css/amp.css]: Please add slightly more comments here. E.g. "authors can set this class on <html> element to ..."

**Comment by dvoytenko** [comment]
@cogell All looks good. I have two minor comments. And, yes, please do add a section to the `docs/inlude_features.md`

File: css/amp.css (hunk 2 above)
Comment (cogell) [inline css/amp.css]: :+1:

**Comment by cogell** [comment]
on it...

**Comment by cogell** [comment]
@dvoytenko updated!

**Comment by jridgewell** [comment]
Is there a reason to restrict this to `html`?

**Comment by cogell** [comment]
hm! sure it seems like this would be fine too:
```
.amp-border-box,
.amp-border-box *,
.amp-border-box *:before,
.amp-border-box *:after {
  box-sizing: border-box
}
```

**Comment by ampsauce** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 27b6b9d5b29038d70f192dba65c0eccff40d387d
Build details: https://travis-ci.org/ampsauce/amphtml/builds/105272992

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cogell** [comment]
@jridgewell updated

**Comment by ampsauce** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: d19e56c46fecb5a50c17f19a50809b672ec2b2bd
Build details: https://travis-ci.org/ampsauce/amphtml/builds/105272873

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by ampsauce** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 18bcd2d9565f85e5817623c36adf6f34764f49d2
Build details: https://travis-ci.org/ampsauce/amphtml/builds/105278263

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: docs/include_features.md [hunk 4]
```diff
@@ -241,6 +241,11 @@ if the browser doesn't support HTML5 audio, for example:
   <source type="audio/ogg" src="foo.ogg">
 </amp-audio>
 ```
+# Add Border Box Sizing
+
+Included in the base amp css is a class of `amp-border-box` that will set `box-sizing: border-box` on all elements
```
Comment (dvoytenko) [inline docs/include_features.md]: "amp css" -> "AMP CSS"

**Comment by dvoytenko** [comment]
Great! LGTM with one minor comment.

**Comment by dvoytenko** [comment]
Merged. Thanks a lot!

**Comment by azizhk** [comment]
Please guide me with this PR:
https://github.com/ampproject/amphtml/pull/3360

It adds box-sizing to all elements but it can be easily overriden.
